test/e2e/Services/DockerContainer.spec.ts

```diff
@@ -15,59 +14,85 @@ class SimpleDockerContainer extends DockerContainer {
 }
 
 describe('Docker container e2e tests', () => {
-    let dockerService: DockerContainer;
+    let dockerContainer: DockerContainer;
 
-    beforeEach(async () => {
-        dockerService = new SimpleDockerContainer();
+    beforeEach(() => {
+        dockerContainer = new SimpleDockerContainer();
     });
 
     // clean up created container
     afterEach(async () => {
-        await dockerService.stopDockerInstance();
-        await runCmd(`docker rm test-image`);
+        const isRunning = await dockerContainer.isRunning();
+        if (isRunning) {
+            await dockerContainer.stop();
+        }
+        await runCmdAsync(`docker rm test-image`).catch();
     }, 20000);
 
-    it('should run docker instance', async () => {
+    /**
+     * TEST CASE: run-check
+     * 1) run docker instance
+     * 2) check if instance is running using cmdRun
+     */
+    it('should execute test case: run-check', async () => {
         if (await DockerContainer.isDockerInstalled()) {
-            const docker = await dockerService.run();
-            expect(docker.running).toBeTruthy();
-            const res = await runCmd(DockerCommand.ps('test-image', 'running'));
-            if (!(res.stdout instanceof Readable)) {
-                expect(res.stdout.split('\n')[1] !== '').toBeTruthy();
-            }
-            const stopInstance = await dockerService.stopDockerInstance();
-            expect(stopInstance).toBeTruthy();
-            expect(await dockerService.isDockerInstanceRunning()).toBeFalsy();
+            await dockerContainer.run();
+            // wait for docker instance to start
+            while (!(await dockerContainer.isRunning())) {}
+
+            const res = await runCmdAsync(DockerCommand.ps('test-image', 'running'));
+            expect(res.stdout.split('\n')[1] !== '').toBeTruthy();
```

Comment:
Why are you testing runCmdAsync here? If you are testing docker container, just pass test if passes while loop